packages/integrations/timer/src/server.ts

```diff
@@ -0,0 +1,21 @@
+import { polyfill } from '@astrojs/webapi';
+import type { IncomingMessage, ServerResponse } from 'http';
+import type { SSRManifest } from 'astro';
+import { NodeApp } from 'astro/app/node';
+
+polyfill(globalThis, {
+	exclude: 'window document',
+});
+
+export function createExports(manifest: SSRManifest) {
+	const app = new NodeApp(manifest);
+	return {
+		handler: async (req: IncomingMessage, res: ServerResponse) => {
+			const start = performance.now();
+			await app.render(req);
+			const end = performance.now();
+			res.write(end - start + '');
+			res.end();
+		},
```

Comment:
The main code in the `@astrojs/timer` integration is here. The rest are largely copied from `@astrojs/node` but stripped out the unneeded parts.